components/SigninForm.js

```diff
@@ -1,83 +1,138 @@
 import styled from "styled-components";
+import {useState, useEffect} from "react";
+
+export default function SigninForm({showPassword, onShowPassword, onSignIn}) {
+  const [users, setUsers] = useState([]);
+  const [loginFilter, setLoginFilter] = useState({
+    firstName: "",
+    lastName: "",
+    password: "",
+  });
+  const [loginFailed, setLoginFailed] = useState(false);
+
+  const handleLoginFailed = () => {
+    setLoginFailed(true);
+  };
+
+  useEffect(() => {
+    const getUsers = async () => {
+      try {
+        const url =
+          loginFilter.length === 0
```

Comment:
This won't work as `loginFilter` is an object which has no property length... Did you mean `loginFilter.firstName.length`?

Reply:
Yes, I forgot to add `.firstName`. 🙈  I have corrected it now...